iothub_client/src/iothub_client_authorization.c

```diff
@@ -666,3 +667,34 @@ char* IoTHubClient_Auth_Get_TrustBundle(IOTHUB_AUTHORIZATION_HANDLE handle, cons
     return result;
 }
 #endif
+
+int IoTHubClient_Auth_Set_SasToken_Expiry(IOTHUB_AUTHORIZATION_HANDLE handle, size_t expiry_time_seconds)
+{
+    int result;
+    if (handle == NULL)
+    {
+        LogError("Invalid handle value handle: NULL");
+        result = __FAILURE__;
+    }
+    else
+    {
+        handle->token_expiry_time_sec = expiry_time_seconds;
+        result = 0;
+    }
+    return result;
+}
+
+size_t IoTHubClient_Auth_Get_SasToken_Expiry(IOTHUB_AUTHORIZATION_HANDLE handle)
+{
+    size_t result;
+    if (handle == NULL)
+    {
+        LogError("Invalid handle value handle: NULL");
+        result = 0;
```

Comment:
p2 comment...
So 0 is both expired and error. Should we separate?

Reply:
We can't, on error you can't send back anything else.  We are just checking the error so we don't segfault.

---
In reply to: [251145143](https://github.com/Azure/azure-iot-sdk-c/pull/825#discussion_r251145143) [](ancestors = 251145143)